state-chain/pallets/cf-witnesser/src/lib.rs

```diff
@@ -495,7 +497,32 @@ pub mod pallet {
 			call: Box<<T as Config>::RuntimeCall>,
 		) -> DispatchResult {
 			T::EnsureWitnessed::ensure_origin(origin)?;
-			Self::deposit_event(Event::<T>::Prewitnessed { call: *call });
+			Self::deposit_event(Event::<T>::Prewitnessed { call: *call.clone() });
+			Ok(())
+		}
+
+		/// Emits an event to notify that this call has been witnessed. Then, it dispatches the call
+		/// using the prewitness threshold origin.
+		#[pallet::call_index(3)]
+		#[pallet::weight(call.get_dispatch_info().weight)] // TODO JAMIE: weight
+		pub fn prewitness_and_execute(
+			origin: OriginFor<T>,
+			call: Box<<T as Config>::RuntimeCall>,
```

Comment:
initially thought there might have been an issue here, but there's not - however still think it would be a good idea to add a test for this case (could be in a separate PR). The case being: 

A call is `prewitness_and_execute`d successfully. And then it's witnessed successfully. 

The reason for testing this explicitly is to ensure that we don't, for example, store the `CallHash` as executed after the prewitness goes through, and then we can't execute the main witness call.

Reply:
I thought the test was easy, but i ran into an issue with the origin. Yeah might leave it for another PR. probs a few more tests we can do.